valispace/__init__.py

```diff
@@ -110,23 +110,199 @@ def get_vali(self, id=None, name=None):
 				print("VALISPACE-ERROR: Vali id must be an integer")
 				return
 
+		if workspace_id:
+			try:
+				workspace_id = int(workspace_id)
+			except:
+				print("VALISPACE-ERROR: Workspace id must be an integer")
+				return
+			
+		if project_id:
+			try:
+				project_id = int(project_id)
+			except:
+				print("VALISPACE-ERROR: Project id must be an integer")
+				return
+
+		if parent_id:
+			try:
+				parent_id = int(parent_id)
+			except:
+				print("VALISPACE-ERROR: Parent id must be an integer")
+				return
+
+		if tag_id:
+			try:
+				tag_id = int(tag_id)
+			except:
+				print("VALISPACE-ERROR: Tag id must be an integer")
+				return
+
+		if valis_marked_as_impacted:
+			try:
+				valis_marked_as_impacted = int(valis_marked_as_impacted)
+			except:
+				print("VALISPACE-ERROR: Valis_marked_as_impacted must be an integer")
+				return
+
 		# Check if no argument was passed
-		if id is None and name is None:
-			print("VALISPACE-ERROR: 1 argument expected (name or id)")
+		if id is None and name is None and workspace_id is None and project_id is None and parent_id is None and parent_name is None and tag_id is None and tag_name is None and valis_marked_as_impacted is None:
+			print("VALISPACE-ERROR: at least 1 argument expected")
 			return
 
-		# Check if name argument was passed
-		if id is None:
-			id = self.__name_to_id(name)
+		# Increment function to add multiple fields to url
+		def increment(url):
+			if not url.endswith('?'):
+					url += "&"
+			return url
+
+		# URL
+		url = self.valispace_login['url'] + "vali/?"
+		if id:
+			url += "id=" + str(id)
+		elif name:
+			url += "shortname=" + str(name)
```

Comment:
Shouldn't this be name= instead of shortname? At least that's how it's defined in the readme. Shortname might be a better argument for the filter function.